librz/bin/p/bin_sms.c

```diff
@@ -12,29 +12,64 @@ typedef struct gen_hdr {
 	ut8 RegionRomSize; // Low 4 bits RomSize, Top 4 bits Region
 } SMS_Header;
 
-static ut32 cb = 0;
+typedef struct sms_info {
+	SMS_Header *hdr;
+	ut32 offset;
+} SMS_INFO;
 
-static bool check_buffer(RzBuffer *b) {
+static bool find_magic(RzBuffer *b, ut32 *offset) {
 	ut32 *off, offs[] = { 0x2000, 0x4000, 0x8000, 0x9000, 0 };
 	ut8 signature[8];
 	for (off = (ut32 *)&offs; *off; off++) {
```

Comment:
While you are at it, can you pplease simplify this for loop. This pointer arithmetic hurts my eyes.

Just do it more simple with indices
```c
for (size_t i = 0; i < RZ_ARRAY_SIZE(offs); i++) {
  off = offs[i];
  ...
}
```

Also, please be so kind and rename `offs -> offsets`.